CMakeLists.txt

```diff
@@ -17,45 +19,74 @@ if("$ENV{ARCH_NAME}" STREQUAL "wormhole_b0")
     set(NORMALIZED_ARCH_NAME "wormhole")
 endif()
 
+if(DEFINED ENV{CMAKE_C_COMPILER} AND DEFINED ENV{CMAKE_CXX_COMPILER})
+    message(STATUS "Setting C and C++ compiler from environment variables")
+    set(CMAKE_C_COMPILER $ENV{CMAKE_C_COMPILER})
+    set(CMAKE_CXX_COMPILER $ENV{CMAKE_CXX_COMPILER})
+endif()
+
+if(CMAKE_CXX_COMPILER AND CMAKE_C_COMPILER)
+    message(STATUS "Using specifed C++ compiler: ${CMAKE_CXX_COMPILER}")
+    message(STATUS "Using specifed C compiler: ${CMAKE_C_COMPILER}")
+else()
+    message(STATUS "No C or C++ compiler specified, defaulting to Clang-17")
+    FIND_AND_SET_CLANG17()
+endif()
+
+list(PREPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+set(ENV{CPM_SOURCE_CACHE} "${PROJECT_SOURCE_DIR}/.cpmcache")
+include(CPM)
+
+if(CMAKE_VERSION VERSION_LESS 3.25)
+    add_subdirectory(dependencies EXCLUDE_FROM_ALL)
+else()
+    add_subdirectory(dependencies EXCLUDE_FROM_ALL SYSTEM)
+endif()
+
 message($ENV{TT_METAL_HOME}/tt_metal/hw/inc/${NORMALIZED_ARCH_NAME})
 add_executable(ttnn-hello ttnn-hello.cpp)
 
 target_include_directories(ttnn-hello PRIVATE
-    # Metalium
     $ENV{TT_METAL_HOME}
     $ENV{TT_METAL_HOME}/tt_metal
     $ENV{TT_METAL_HOME}/tt_metal/third_party/umd
     $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/device
-    $ENV{TT_METAL_HOME}/tt_metal/third_party/fmt
-    $ENV{TT_METAL_HOME}/tt_metal/hw/inc/${NORMALIZED_ARCH_NAME}
+    $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/device/api/
+    $ENV{TT_METAL_HOME}/tt_metal/hw/inc/${NORMALIZED_ARCH_NAME}/
     $ENV{TT_METAL_HOME}/tt_metal/hw/inc/
     $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/src/firmware/riscv/${NORMALIZED_ARCH_NAME}
 
     # TTNN
     $ENV{TT_METAL_HOME}/ttnn/cpp
     $ENV{TT_METAL_HOME}/ttnn/cpp/ttnn/deprecated
     $ENV{TT_METAL_HOME}/tt_metal/third_party/magic_enum
-
 )
+
 target_link_directories(ttnn-hello PRIVATE
     $ENV{TT_METAL_HOME}/build/lib
 )
 
-# TODO: In the future TTNN can optionally not depend on Python
-find_package(Python REQUIRED Development)
 target_link_libraries(ttnn-hello PRIVATE
-    # Metalium
-    tt_metal
+    fmt
+    magic_enum
+    Reflect::Reflect
     yaml-cpp
-    pthread
-
-    # TTNN
-    Python::Python
+    Boost::core
+    Boost::container
+    tt_metal
     $ENV{TT_METAL_HOME}/build/lib/_ttnn.so
+    $ENV{TT_METAL_HOME}/build/lib/libdevice.so
+    $ENV{TT_METAL_HOME}/build/lib/libtt_metal.so
 )
 
+target_compile_options(ttnn-hello PRIVATE $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-stdlib=libc++>)
```

Comment:
I recommend add a `USE_LIBCPP` flag instead for forcing libc++ when clang is detected. Clang works perfectly with stdlibc++ on Arch and Metal built on Arch.